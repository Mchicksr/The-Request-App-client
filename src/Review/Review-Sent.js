import React from 'react'
import './Review.css'

function ReviewSent(){
    return(
        <div>
             <h1 className="neon">THANK YOU!</h1><br></br>
        <h1 className="">The DJ Has Your Review!</h1>
        </div>
    )
}

export default ReviewSent;