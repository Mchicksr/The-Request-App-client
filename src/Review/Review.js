import React, {Component} from 'react';
import './Review.css'

class Review extends Component{

    constructor(props) {
        super(props);
        this.state = {
            review:""
        }
    }
    reviewChanged(e) {
        this.setState({
            review: e.target.value
        })
    }

    handleSubmit = (e) => {
        e.preventDefault()
        const { review } = this.state;
        const newReview = { review }
        const url = 'https://mighty-temple-37477.herokuapp.com/api/review'
        const opt ={
            method: 'post',
            body: JSON.stringify(newReview),
            headers: {
                'content-Type': 'application/json'
            }
        }

        fetch(url,opt)
        .then(res => {
            if(!res.ok) {
                throw new Error('something went wrong please try again')
            }
            console.log(res)
            return res.json()
        })
        .then(newReview => {
            this.setState({
                newReview:" "
            })
            this.props.handleAdd(newReview)
        })
        .catch(err => {
            this.setState({
                error:err.message
            })
            window.location.href='/review-sent'
        })

    }

    onSubmit(e){
        e.preventDefault()
    }
    render(){
    return(
        <div>
        <h1 className="neon">Review</h1>
    <h2>How was your expreience?</h2>
   
    <form
    
    onSubmit={e => this.handleSubmit(e)} >
         <label for="review">Review</label><br></br>
        <textarea 
        
        value={this.state.review}
        onChange={e => this.reviewChanged(e)}
        placeholder="comment">
        </textarea><br></br>
        <button  
        disabled={!this.state.review }
        type='submit'
        className= "req" >
            Send Review
        </button>
    </form>
    </div>
    )
    }
}

export default Review;