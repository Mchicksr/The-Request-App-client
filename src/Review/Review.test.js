import React from 'react'
import Review from './Review'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import ReactDOM from 'react-dom'
import Adapter from 'enzyme-adapter-react-16'
configure({adapter:new Adapter()})
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'


it('renders without crashing,',()=>{
    const div = document.createElement('div');
    ReactDOM.render( <BrowserRouter ><Review /></BrowserRouter>, div)
    })


it("matches snapshot review", () => {
    // const tree = renderer.create(<Button label="click me please"></Button>).toJSON();
      const tree = renderer.create(  <a class="req" href="/Review"> Send Request </a>).toJSON();
  
    expect(tree).toMatchSnapshot();
  })