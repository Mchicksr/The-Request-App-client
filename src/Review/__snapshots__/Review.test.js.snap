// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot review 1`] = `
<a
  class="req"
  href="/Review"
>
   Send Request 
</a>
`;
